src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilderResource.java

```diff
@@ -0,0 +1,175 @@
+        DONT_HAVE,
+        NEED_MORE,
+        HAVE_ENOUGH
+    }
+
+    private       int       amountAvailable;
+    private       int       amountPlayer;
+
+
+    /**
+     * Constructor for a resource.
+     *
+     * @param item              the item.
+     * @param damageValue       it's damage value.
+     * @param amount            amount for this resource.
+     */
+     public BuildingBuilderResource(@NotNull final Item item, final int damageValue, final int amount)
+     {
+         super(item, damageValue, amount, false);
+         this.amountAvailable=0;
+         this.amountPlayer=0;
+     }
+
+    /**
+     * Constructor for a resource.
+     *
+     * @param item              the item.
+     * @param damageValue       it's damage value.
+     * @param amount            amount for this resource.
+     * @param available         optional amount available for this resource
+     */
+     public BuildingBuilderResource(@NotNull final Item item, final int damageValue, final int amount, final int available)
+     {
+         this(item, damageValue, amount);
+         this.amountAvailable=available;
+     }
+
+    public String getName()
+    {
+        //It is the bet way ?
+        return new ItemStack(getItem()).getDisplayName();
+    }
+
+    /**
+     * get the amount available for this resource.
+     * i.e. amount in the chest + amount in the builder's inventory
+     */
+    public int getAvailable()
+    {
+        return amountAvailable;
+    }
+
+    /**
+     * Setter for the available resource amount.
+     *
+     * @param amount this is the new amount available
+     */
+    public void setAvailable(final int amount)
+    {
+        amountAvailable = amount;
+    }
+
+    /**
+     * Add to the current amount available.
+     *
+     * @param amount to add to the current amount available
+     */
+    public void addAvailable(final int amount)
+    {
+        this.amountAvailable+=amount;
+    }
+
+    public void setPlayerAmount(final int amount)
+    {
+        amountPlayer = amount;
+    }
+
+    public int getPlayerAmount()
+    {
+        return amountPlayer;
+    }
+
+    public RessourceAvailability getAvailabilityStatus()
+    {
+        if (getAmount() > amountAvailable)
+        {
+            if (amountPlayer == 0)
+            {
+                return RessourceAvailability.DONT_HAVE;
+            }
+            if (amountPlayer < (getAmount()-amountAvailable))
+            {
+                return RessourceAvailability.NEED_MORE;
+            }
+            return RessourceAvailability.HAVE_ENOUGH;
+        }
+        return RessourceAvailability.NOT_NEEDED;
+    }
+
+    @Override
+    public String toString()
+    {
+        final int itemId= Item.getIdFromItem(getItem());
+        return getName() + "(p:"+amountPlayer+" a:" +amountAvailable+" n:"+getAmount()+" id="+itemId+" damage="+getDamageValue()+") => "+getAvailabilityStatus().name();
+    }
+
+    @Override
+    public boolean equals(final Object o)
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') This class overrides "equals()" and should therefore also override "hashCode()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1206)